packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -1752,6 +1736,36 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
     }
     return this._connections;
   }
+
+
+  /**
+   * Set the domain access policies associated with the domain.
+   */
+  public addAccessPolicies(accessPolicyStatements: iam.PolicyStatement[]) {
+    if (this.accessPolicy) throw new Error('The access policy can only be set once');
+
+    this.accessPolicy = new ElasticsearchAccessPolicy(this, 'ESAccessPolicy', {
+      domainName: this.domainName,
+      domainArn: this.domainArn,
+      accessPolicies: accessPolicyStatements,
+    });
+
+    if (this.encryptionAtRest?.kmsKey) {
+      // https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/encryption-at-rest.html
+
+      // these permissions are documented as required during domain creation.
+      // while not strictly documented for updates as well, it stands to reason that an update
+      // operation might require these in case the cluster uses a kms key.
+      // empircal evidence shows this is indeed required: https://github.com/aws/aws-cdk/issues/11412
+      this.accessPolicy.grantPrincipal.addToPrincipalPolicy(new iam.PolicyStatement({
+        actions: ['kms:List*', 'kms:Describe*', 'kms:CreateGrant'],
+        resources: [this.encryptionAtRest.kmsKey.keyArn],
+        effect: iam.Effect.ALLOW,
+      }));
+    }
+
+    this.accessPolicy.node.addDependency(this.domain);
```

Comment:
should be created automatically by passing `domainName`, `domainArn`, and any access policies that contain such values

Reply:
It seems this was done because `this.domainName` and `this.domainArn` don't establish the DependsOn relationship.

I even tried it with `this.domain.domainName`, but that still resulted in this integration test change: 

```Resources
[~] AWS::IAM::Policy DomainESAccessPolicyCustomResourcePolicy9747FC42 
 └─ [-] DependsOn
     └─ ["Domain66AC69E0"]
[~] Custom::ElasticsearchAccessPolicy DomainESAccessPolicy89986F33 
 └─ [~] DependsOn
     └─ @@ -1,4 +1,3 @@
        [ ] [
        [-]   "DomainESAccessPolicyCustomResourcePolicy9747FC42",
        [-]   "Domain66AC69E0"
        [+]   "DomainESAccessPolicyCustomResourcePolicy9747FC42"
        [ ] ]
```

Keeping the explicit dependency is the only way I could make it pass.

Reply:
It is not necessary to have an explicit DependsOn clause if the resource properties contain an explicit reference to the dependent resource. This means that since the definition of `DomainESAccessPolicy89986F33` contains a `{ Ref: Domain66AC69E0 }`, there is no need for an explicit dependency using DependsOn

Reply:
Ok, I switched back to using `this.domain.attrArn` to ensure the implicit dep exists. The integration tests have been updated to reflect.